prediction_market_agent/agents/arbitrage_agent/deploy.py

```diff
@@ -0,0 +1,211 @@
+import typing as t
+
+from langchain_core.output_parsers import PydanticOutputParser
+from langchain_core.prompts import PromptTemplate
+from langchain_core.runnables import RunnableSerializable
+from langchain_openai import ChatOpenAI
+from prediction_market_agent_tooling.deploy.agent import (
+    MAX_AVAILABLE_MARKETS,
+    DeployableTraderAgent,
+)
+from prediction_market_agent_tooling.gtypes import Probability
+from prediction_market_agent_tooling.loggers import logger
+from prediction_market_agent_tooling.markets.agent_market import (
+    AgentMarket,
+    FilterBy,
+    SortBy,
+)
+from prediction_market_agent_tooling.markets.data_models import (
+    BetAmount,
+    Position,
+    ProbabilisticAnswer,
+    TokenAmount,
+    Trade,
+    TradeType,
+)
+from prediction_market_agent_tooling.markets.markets import MarketType
+from prediction_market_agent_tooling.markets.omen.omen import OmenAgentMarket
+from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import (
+    OmenSubgraphHandler,
+)
+from prediction_market_agent_tooling.tools.langfuse_ import (
+    get_langfuse_langchain_config,
+    observe,
+)
+from prediction_market_agent_tooling.tools.utils import utcnow
+
+from prediction_market_agent.agents.arbitrage_agent.data_models import (
+    CorrelatedMarketPair,
+    Correlation,
+)
+from prediction_market_agent.agents.arbitrage_agent.prompt import PROMPT_TEMPLATE
+from prediction_market_agent.db.pinecone_handler import PineconeHandler
+from prediction_market_agent.utils import APIKeys
+
+
+class DeployableArbitrageAgent(DeployableTraderAgent):
+    """Agent that places mirror bets on Omen for (quasi) risk-neutral profit."""
+
+    model = "gpt-4o"
+    # trade amount will be divided between correlated markets.
+    total_trade_amount = BetAmount(amount=0.1, currency=OmenAgentMarket.currency)
+    bet_on_n_markets_per_run = 5
+    max_related_markets_per_market = 10
+
+    def run(self, market_type: MarketType) -> None:
+        if market_type != MarketType.OMEN:
+            raise RuntimeError(
+                "Can arbitrage only on Omen since related markets embeddings available only for Omen markets."
+            )
+        self.subgraph_handler = OmenSubgraphHandler()
+        self.pinecone_handler = PineconeHandler()
+        self.pinecone_handler.update_markets()
+        self.chain = self._build_chain()
+        super().run(market_type=market_type)
+
+    def get_markets(
+        self,
+        market_type: MarketType,
+        limit: int = MAX_AVAILABLE_MARKETS,
+        sort_by: SortBy = SortBy.CLOSING_SOONEST,
+        filter_by: FilterBy = FilterBy.OPEN,
+    ) -> t.Sequence[AgentMarket]:
+        return super().get_markets(
+            market_type=market_type,
+            limit=50,
```

Comment:
`get_markets` accepts the `limit` argument, but it's overruled by this. Not a nice design.

```suggestion
            limit=limit,
```

Reply:
I understand your concern (expressed in 4 separate comments) - 50 is not limit, we should use limit.
So what's the recommended way to provide a custom `limit` apart from overriding the method `process_markets`?

Reply:
For example, you could change the definition of `get_markets` (in PMAT) from

```
 def get_markets(
        self,
        market_type: MarketType,
        limit: int = MAX_AVAILABLE_MARKETS,
        sort_by: SortBy = SortBy.CLOSING_SOONEST,
        filter_by: FilterBy = FilterBy.OPEN,
    ) -> t.Sequence[AgentMarket]:
``` to just `def get_markets(self) -> t.Sequence[AgentMarket]`, and then it's fine to overload the method like this.

Or you could introduce a new configuration variable, `n_markets_to_fetch,` similarly as we have `bet_on_n_markets_per_run.`

Or, like we have BettingStrategy, we could have GetMarketsStrategy.

No strong preferences from my side here.

Reply:
> Or you could introduce a new configuration variable, n_markets_to_fetch, similarly as we have bet_on_n_markets_per_run.

Me and @evangriffiths also like this. Implementing this on PMAT and referencing the new release on this PR afterwards.

Reply:
Implementing this here on PMAT -> https://github.com/gnosis/prediction-market-agent-tooling/pull/532/files

Reply:
Implemented, removing `get_markets`. This thread can be closed IMO.